cmd/util/cmd/rollback-executed-height/cmd/rollback_executed_height.go

```diff
@@ -0,0 +1,277 @@
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not roll back executed block at height %v", flagHeight)
+	}
+
+	log.Info().Msgf("executed height rolled back to %v", flagHeight)
+
+}
+
+// use badger instances directly instead of stroage interfaces so that the interface don't
+// need to include the Remove methods
+func removeExecutionResultsFromHeight(
+	protoState protocol.State,
+	headers *badger.Headers,
+	transactionResults *badger.TransactionResults,
+	commits *badger.Commits,
+	chunkDataPacks *badger.ChunkDataPacks,
+	results *badger.ExecutionResults,
+	myReceipts *badger.MyExecutionReceipts,
+	events *badger.Events,
+	serviceEvents *badger.ServiceEvents,
+	fromHeight uint64) error {
+	log.Info().Msgf("removing results for blocks from height: %v", fromHeight)
+
+	root, err := protoState.Params().Root()
+	if err != nil {
+		return fmt.Errorf("could not get root: %w", err)
+	}
+
+	if fromHeight <= root.Height {
+		return fmt.Errorf("can only remove results for block above root block. fromHeight: %v, rootHeight: %v", fromHeight, root.Height)
+	}
+
+	final, err := protoState.Final().Head()
+	if err != nil {
+		return fmt.Errorf("could get not finalized height: %w", err)
+	}
+
+	if fromHeight > final.Height {
+		return fmt.Errorf("could not remove results for unfinalized height: %v, finalized height: %v", fromHeight, final.Height)
+	}
+
+	finalRemoved := 0
+	total := int(final.Height-fromHeight) + 1
+
+	// removing for finalized blocks
+	for height := fromHeight; height <= final.Height; height++ {
+		head, err := protoState.AtHeight(height).Head()
+		if err != nil {
+			return fmt.Errorf("could not get header at height: %w", err)
+		}
+
+		blockID := head.ID()
+
+		err = removeForBlockID(headers, commits, transactionResults, results, chunkDataPacks, myReceipts, events, serviceEvents, blockID)
+		if err != nil {
+			return fmt.Errorf("could not remove result for finalized block: %v, %w", blockID, err)
+		}
+
+		finalRemoved++
+		log.Info().Msgf("result at height %v has been removed. progress (%v/%v)", height, finalRemoved, total)
+	}
+
+	// removing for pending blocks
+	pendings, err := protoState.Final().Descendants()
+	if err != nil {
+		return fmt.Errorf("could not get pending block: %w", err)
+	}
+
+	pendingRemoved := 0
+	total = len(pendings)
+
+	for _, pending := range pendings {
+		err = removeForBlockID(headers, commits, transactionResults, results, chunkDataPacks, myReceipts, events, serviceEvents, pending)
+
+		if err != nil {
+			return fmt.Errorf("could not remove result for pending block %v: %w", pending, err)
+		}
+
+		pendingRemoved++
+		log.Info().Msgf("result for pending block %v has been removed. progress (%v/%v) ", pending, pendingRemoved, total)
+	}
+
+	log.Info().Msgf("removed height from %v. removed for %v finalized blocks, and %v pending blocks",
+		fromHeight, finalRemoved, pendingRemoved)
+
+	return nil
+}
+
+// removeForBlockID remove block execution related data for a given block.
+func removeForBlockID(
```

Comment:
I have addressed all the review comments. 

The `removeForBlockID` will remove all the results related data including events, serviceEvents, etc. 

And I've verified with a test case that save the results data, and remove it, and then save again. Also tested unhappy path that save again with a different result to simulate the case where there is an execution fork.

Tests are passing. 

Also tested with localnet, rollback to a height, and after starting backup, the node is able to re-execute blocks, and continue executing new blocks. 

cc @AlexHentschel